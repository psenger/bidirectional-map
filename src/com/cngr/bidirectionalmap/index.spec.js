const BiDirectionalMap = require('./index');

describe('BiDirectionalMap',()=>{
    test('#constructor',()=>{
        const bidiMap =  new BiDirectionalMap([['bob', 'larry'],['tim', 'tom']]);
        expect(bidiMap).toMatchSnapshot();
    });
    test('#set',()=>{
       const bidiMap =  new BiDirectionalMap();
        bidiMap.set('bob', 'larry').set('tim', 'tom')
        expect(bidiMap).toMatchSnapshot();
    })
  test('#set with override',()=>{
    const bidiMap =  new BiDirectionalMap();
    bidiMap.set('bob', 'larry').set('bob', 'kim')
    expect(bidiMap).toMatchSnapshot();
  })
})
