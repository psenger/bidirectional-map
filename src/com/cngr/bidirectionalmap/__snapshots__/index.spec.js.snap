// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BiDirectionalMap #constructor 1`] = `
Map {
  "larry" => "bob",
  "bob" => "larry",
  "tom" => "tim",
  "tim" => "tom",
}
`;

exports[`BiDirectionalMap #set 1`] = `
Map {
  "bob" => "larry",
  "larry" => "bob",
  "tim" => "tom",
  "tom" => "tim",
}
`;

exports[`BiDirectionalMap #set with override 1`] = `
Map {
  "bob" => "kim",
  "kim" => "bob",
}
`;
